src/explore-education-statistics-common/src/modules/table-tool/utils/applyTableHeadersOrder.ts

```diff
@@ -0,0 +1,439 @@
+  groups: Filter[][],
+  groupToMatch: Filter[],
+): Filter[] | undefined => {
+  const groupToMatchFilterType = getFilterType(groupToMatch[0]);
+  return groups.find(
+    group => getFilterType(group[0]) === groupToMatchFilterType,
+  );
+};
+
+/**
+ * Order items within a group based on the previous reordering.
+ * Items that weren't in the reordered list will be after the reordered ones.
+ */
+const orderItems = (
+  reorderedItems: Filter[],
+  updatedItems: Filter[],
+): Filter[] => {
+  return reorderedItems
+    .filter(item =>
+      updatedItems.find(updatedItem => updatedItem.value === item.value),
+    )
+    .concat(
+      updatedItems.filter(
+        updatedItem =>
+          !reorderedItems.find(item => item.value === updatedItem.value),
+      ),
+    );
+};
+
+const isReorderedGroup = (
+  originalGroup: Filter[],
+  updatedGroup: Filter[],
+): boolean => {
+  const intersectionOrderedByOriginal = intersectionWith(
+    originalGroup,
+    updatedGroup,
+    isEqual,
+  );
+  const intersectionOrderedByUpdated = intersectionWith(
+    updatedGroup,
+    originalGroup,
+    isEqual,
+  );
+  return !isEqual(intersectionOrderedByOriginal, intersectionOrderedByUpdated);
+};
+
+/**
+ * Have the groups in an axis been re-ordered or moved between axes
+ * */
+const areGroupsReordered = (
+  previousOrder: string[],
+  previousOtherAxisOrder: string[],
+  updatedOrder: string[],
+  updatedOtherAxisOrder: string[],
+) => {
+  if (isEqual(previousOrder, updatedOrder)) {
+    return false;
+  }
+
+  const intersectionOrderedByPrevious = intersection(
+    previousOrder,
+    updatedOrder,
+  );
+  const intersectionOrderedByUpdated = intersection(
+    updatedOrder,
+    previousOrder,
+  );
+  if (!isEqual(intersectionOrderedByPrevious, intersectionOrderedByUpdated)) {
+    return true;
+  }
+  const anyGroupsMovedOutOfAxis = previousOrder.find(group =>
+    updatedOtherAxisOrder.includes(group),
+  );
+  const anyGroupsMovedIntoAxis = updatedOrder.find(group =>
+    previousOtherAxisOrder.includes(group),
+  );
+
+  if (anyGroupsMovedOutOfAxis || anyGroupsMovedIntoAxis) {
+    return true;
+  }
+  return false;
+};
+
+const getAdditionalGroups = (
+  otherAxisReorderedGroups: Filter[][],
+  reorderedGroups: Filter[][],
+  updatedGroups: Filter[][],
+) => {
+  return updatedGroups.filter(group => {
+    const matchingGroupInAxis = getMatchingGroup(reorderedGroups, group);
+    const matchingGroupInOtherAxis = getMatchingGroup(
+      otherAxisReorderedGroups,
+      group,
+    );
+    return !matchingGroupInAxis && !matchingGroupInOtherAxis;
+  });
+};
+
+/**
+ * Filter the reordered groups by ones that exist in either axis of the updated groups.
+ * If the group items were reordered, or were reordered in the original saved version, then maintain that order, if not use the default order.
+ */
+const filterAndOrderReorderedGroups = ({
+  allOriginalGroups,
+  reorderedGroups,
+  updatedGroups,
+  updatedGroupsOnOtherAxis,
```

Comment:
Similar to my suggestion about renaming `updatedTableHeaders`' , we could probably rename these to `defaultGroups` and `otherAxisDefaultGroups` to be a little bit clearer what they mean.

Would make this change for other references to `updatedGroup` and `updatedItem` in other functions as well.

Reply:
done